src/deps_info.json

```diff
@@ -94,5 +94,8 @@
   "_embind_register_std_string": ["malloc", "free"],
   "_embind_register_std_wstring": ["malloc", "free"],
   "__syscall192": ["emscripten_builtin_memalign"],
-  "pthread_create": ["malloc", "free", "emscripten_main_thread_process_queued_calls"]
+  "pthread_create": ["malloc", "free", "emscripten_main_thread_process_queued_calls"],
+  "$getTypeName": ["free"],
+  "_embind_register_std_string": ["free"],
+  "_embind_register_std_wstring": ["free"]
```

Comment:
@kripken: For some reason I still can't get my head around the deps_info.json behavior. :/ 

E.g. `other.test_embind` calls `_embind_register_std_string` directly from C/C++ code (`__embind_register_native_and_builtin_types` in `bind.cpp`), and `_embind_register_std_string` depends on `free`, so I put in a dependency `"_embind_register_std_string": ["free"],` here, but even then for some reason the function free is not getting included to the build, hence I need an explicit `shared.Settings.EXPORTED_FUNCTIONS += ['_free']` above.

Reply:
That seems like it should work. I wonder if there is something special about embind linking perhaps?

Should be easy to debug, `add_back_deps` is where the simple logic for deps_info is done, in tools/system_libs.py. Verifying that symbol is undefined, and in the list of symbols scanned there, should show what is going wrong.

Reply:
Still not sure I understand what's going on.. The test program does use `_embind_register_std_string` for example, but it is not present in `undefs`. Adding the dependency to `free` to a few more functions that do also flow a call to `free` resolves the issue. Perhaps that is enough of a coverage to make sure it gets picked up always.

Reply:
Was this resolved? I don't see `_free` added to EXPORTED_FUNCTIONS and the tests seem to pass?

Reply:
Oh yeah, added more functions to deps info, which fixed it.